### PR TITLE
Add cppcodec recipe

### PR DESCRIPTION
**Pull request by ZaMaZaN4iK**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Hi @ZaMaZaN4iK ! Thanks for pushing this PR! Please, consider https://github.com/ZaMaZaN4iK/conan-center-index/pull/1 first. Regards!

**Comment by ZaMaZaN4iK** [comment]
@uilianries can you please trigger a build for the PR? Thank you :)

**Comment by uilianries** [comment]
@ZaMaZaN4iK Unfortunately not, you must be approved on #4 first.

**Comment by ZaMaZaN4iK** [comment]
@uilianries Is it possible some how facilitate a process about merging this recipe into conan-center-index? That's a little bit strange that even with ready recipe it cannot be merged immediately.

**Comment by uilianries** [comment]
@ZaMaZaN4iK Yes, but will take some time. First we need to stabilize the CI service, it's not 100% yet, there are some minor bugs, but we are working on it. Also, we can add a bot to merge when having 2 positive reviews, but probably it will occur only after EAP, because we need a stable CI first, otherwise, we can merge a PR with internal broken packages, or absent package as we saw few times.

File: recipes/cppcodec/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
+
+
+class CppcodecConan(ConanFile):
+    name = "cppcodec"
+    version = "0.2"
```
Comment (uilianries) [inline recipes/cppcodec/all/conanfile.py]:
```suggestion
```
As you are using the generic folder "all", version is not required.

**Comment by ZaMaZaN4iK** [comment]
@danimtb can you please approve this PR for merging? Thank you!

**Comment by ZaMaZaN4iK** [comment]
@SSE4 can you approve the PR? Thank you!

**Pull request merged**